internal/resolver/dns/dns_resolver_test.go

```diff
@@ -1490,55 +1498,63 @@ func TestRateLimitedResolve(t *testing.T) {
 		t.Fatalf("delegate resolver returned unexpected type: %T\n", tr)
 	}
 
-	// Observe the time before unblocking the lookupHost call.  The 100ms rate
-	// limiting timer will begin immediately after that.  This means the next
-	// resolution could happen less than 100ms if we read the time *after*
-	// receiving from tr.ch
-	start := time.Now()
+	ctx, ctxCancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer ctxCancel()
+
 
 	// Wait for the first resolution request to be done. This happens as part
-	// of the first iteration of the for loop in watcher() because we call
-	// ResolveNow in Build.
-	<-tr.ch
-
-	// Here we start a couple of goroutines. One repeatedly calls ResolveNow()
-	// until asked to stop, and the other waits for two resolution requests to be
-	// made to our testResolver and stops the former. We measure the start and
-	// end times, and expect the duration elapsed to be in the interval
-	// {wantCalls*dnsResRate, wantCalls*dnsResRate}
-	done := make(chan struct{})
-	go func() {
-		for {
-			select {
-			case <-done:
-				return
-			default:
-				r.ResolveNow(resolver.ResolveNowOptions{})
-				time.Sleep(1 * time.Millisecond)
-			}
-		}
-	}()
+	// of the first iteration of the for loop in watcher().
+	select {
+	case <-tr.ch:
+	case <-ctx.Done():
+		t.Fatal("Timed out waiting for lookup() call.")
+	}
 
-	gotCalls := 0
-	const wantCalls = 3
-	min, max := wantCalls*dnsResRate, (wantCalls+1)*dnsResRate
-	tMax := time.NewTimer(max)
-	for gotCalls != wantCalls {
-		select {
-		case <-tr.ch:
-			gotCalls++
-		case <-tMax.C:
-			t.Fatalf("Timed out waiting for %v calls after %v; got %v", wantCalls, max, gotCalls)
-		}
+	// Call Resolve Now 100 times, shouldn't continue onto next iteration of watcher, thus shouldn't lookup again.
+	for i := 0; i <= 100; i++ {
+		r.ResolveNow(resolver.ResolveNowOptions{})
+	}
+
+	shouldNotHappenCtx, shouldNotHappenCtxCancel := context.WithTimeout(context.Background(), defaultTestShortTimeout)
+	defer shouldNotHappenCtxCancel()
+
+	select {
+	case <-tr.ch:
+		t.Fatalf("Should not have looked up again as DNS Min Res Rate timer has not gone off.")
+	case <-shouldNotHappenCtx.Done():
+	}
```

Comment:
s/shouldNotHappenCtx/sCtx

In Go, we generally prefer shorter variable names. A good rule of thumb for the length of the variable names is their scope. Shorter the scope, shorter the variable name.
See: go/go-style/decisions#variable-names

Reply:
No wonder there's so many abbreviations. Personally, I hate abbreviations. I spend so much swe time figuring out what abbreviations mean. I disagree with this style guideline.

Reply:
Switching a variable name from explaining what it does to sCtx (what does that mean...) will require some explanation from the "Go Readability" community contributor because I completely disagree with it.

Reply:
Don't use too many abbreviations for things in longer functions.  If the variable is declared and used completely within like 5 lines, a single letter is fine.  Otherwise more descriptive names are usually better.  I dislike `sCtx`, and also `shouldNotHappenCtx`.  Since we already have a `ctx`, how about something like `passCtx` or `continueCtx`?

Reply:
If you are new to Go, shorter variable names could seem very counter intuitive. Long variable names for things which have a scope of a few lines definitely hurt over time. 

There is a long rant about why Go prefers shorter variable names in the link that I sent. On what points do you disagree? And why?

Reply:
Argh, I still like shouldNotHappenCtx the most. I'll switch to "continueCtx" (begrudgingly).

Reply:
I don't know @easwars, it was a (completely personal, and very unique) problem I had with Envoy as well, so many abbreviations. My philosophy about it comes from a book I read called clean code - where it has a section about explaining yourself with variable names. For example, like even if I'm in say a file called ClusterHandler with a ClusterHandler type, having the capture argument being (ch *ClusterHandler), I have to spend a few seconds figuring out what ch.X stands for. It seems to be an extremely unique ideology. I agree with keeping things clean, but I also personally think that variable names should be as verbose as possible.

Reply:
I think what I said mostly agrees with it.  We have problems in our codebase of excessive abbreviation.  E.g.s: `ClientConn.sc/mkp` or `clientStream.cp`.

I wasn't a fan of `sCtx` since it's live for the entire function and what does `s` even mean?  Why not `zCtx`?  AFAIK there is no commonly established convention for contexts like `i` as integer loop counters.  Also, unlike `for v := range vThings`, it is unclear from `sCtx`'s initialization what it will be used for.  You could add a comment, but clearer variable names are better than comments that explain unclear variable names.

`shouldNotHappenCtx` is verbose _and_ potentially misleading.  What "should not happen"?  The context expiration?  Is this something that is triggered _when_ an unwanted thing happens?  Because it sounds like that to me.

Reply:
Fair point about shouldNotHappen. I'll try and be reasonable about balancing verbosity and cleanliness wrt variable naming moving forward.